stdlib/mexpr/eq.mc

```diff
@@ -362,14 +362,44 @@ lang VarPatEq = VarPat
     else None ()
 end
 
-lang SeqTotPatEq
+lang SeqTotPatEq = SeqTotPat
   sem eqPat (env : Env) (free : Env) (patEnv : NameEnv) (lhs : Pat) =
-  -- TODO(dlunde,2020-09-29)
+  | PSeqTot {pats = ps2} ->
+    match lhs with PSeqTot {pats = ps1} then
+      if eqi (length ps2) (length ps1) then
+        let z = zipWith (lam p1. lam p2. (p1,p2)) ps1 ps2 in
+        optionFoldlM (lam fpEnv. lam ps.
+            match fpEnv with (f,p) then
+              eqPat env f p ps.0 ps.1 else None ())
```

Comment:
I believe this should be `else never`, since the pattern should be irrefutable. (there are a couple more places where this is true as well)